loader/windows/icd_windows_dxgk.c

```diff
@@ -37,16 +37,17 @@ bool khrIcdOsVendorsEnumerateDXGK(void)
     bool ret = false;
 #if defined(OPENCL_ICD_LOADER_REQUIRE_WDK)
 #if defined(DXGKDDI_INTERFACE_VERSION_WDDM2_4) && (DXGKDDI_INTERFACE_VERSION >= DXGKDDI_INTERFACE_VERSION_WDDM2_4)
+    // Get handle to GDI Runtime
+    HMODULE h = LoadLibrary("gdi32.dll");
+    if (h && GetProcAddress((HMODULE)h, "D3DKMTSubmitPresentBltToHwQueue")) // OS Version check
```

Comment:
1) Resource leak
    We are not releasing `h` here, we should call FreeLibrary on the handle.
2) Redundant cast to HMODULE in GetProcAddress arg 0
3) Refactoring - there is a better way to express the intent here (IMHO)
I would move these two lines (predicate) into an inline function IsOSVersionGTE_XYZ:
static inline bool IsOSVersionGTE_XYZ(void)
{
    bool ret;
    HMODULE h = LoadLIbrary("gdi32.dll");
    ret = GetProcAddress(h, "D3DKMTSubmitPresentBltToHwQueue") != NULL;
    FreeLibrary(h);
    return;
}

Reply:
I think the gdi32 module is needed even after this check, so freeing it right after OS version check will need to reload it again. I would rather split gdi32 load and D3DKMTSubmitPresentBltToHwQueue query and move the query to a macro.

Reply:
I see this currently only at one place. So, I would defer this for now as there isn't a real gain from refactoring this. I am expecting few changes around this to follow.